gtsam/hybrid/HybridBayesNet.cpp

```diff
@@ -43,4 +47,78 @@ GaussianBayesNet HybridBayesNet::choose(
   return gbn;
 }
 
+/* ************************************************************************* */
+HybridBayesNet::shared_ptr HybridBayesNet::prune(
+    const DecisionTreeFactor::shared_ptr &discreteFactor) const {
+  // To Prune, we visitWith every leaf in the GaussianMixture. For each
+  // leaf, we apply an operation, where using the assignment, we can
+  // check the discrete decision tree for an exception and if yes, then just
+  // set the leaf to a nullptr. We can later check the GaussianMixture for
+  // just nullptrs.
+
+  HybridBayesNet::shared_ptr prunedBayesNetFragment =
```

Comment:
oh no, you are going to modify this, right?? rethink :-)

Reply:
Yup already made the change once you pointed out the `setGaussian` badness.